plugins/imfile/imfile.c

```diff
@@ -1162,7 +1222,11 @@ enqLine(act_obj_t *const act,
 	msgSetPRI(pMsg, inst->iFacility | inst->iSeverity);
 	MsgSetRuleset(pMsg, inst->pBindRuleset);
 	if(inst->addMetadata) {
-		metadata_values[0] = (const uchar*)act->name;
+		if (act->source_name) {
```

Comment:
mhhh... Isn't that changing the existing behavior?

Reply:
No, it is not, this is ensuring that what was changed in the internal structures does *not* affect what user has in metadata.